microsoft/onefuzz
# Improve error reporting from scale-in protection modification

## Summary of the Pull Request

_What is this about?_

Return different error codes from the different failure modes of modifying scale-in protection on a VM instance. Closes #3183 

## Info on Pull Request

_What does this include?_

1. 4 new error codes that represent our known failure modes for modifying scale-in protection
2. Downgrade Upgrade Already In Progress errors to a warning
3. In the case of removing scale-in protection, downgrade Instance No Longer Exists to a warning

## Validation Steps Performed

_How does someone test & validate?_

check-pr should not longer report errors in check logs for modifying scale-in protection
